frontend/dockerui/build.go

```diff
@@ -50,6 +50,23 @@ func (bc *Client) Build(ctx context.Context, fn BuildFunc) (*ResultBuilder, erro
 			if tp != nil {
 				p = *tp
 			}
+
+			// in certain conditions we allow input platform to be extended from base image
+			if p.OS == "windows" && img.OS == p.OS {
```

Comment:
Is there any reason to gate this on Windows?

Reply:
You mean in case other OS would also set `os.version`? I've only seen the field use for windows.